pkgs/development/libraries/libibverbs/default.nix

```diff
@@ -1,17 +1,78 @@
 { stdenv, fetchurl }:
 
-stdenv.mkDerivation rec {
-  name = "libibverbs-1.1.8";
+let
+  
+  verbs = rec {
+      version = "1.1.8";
+      name = "libibverbs-${version}";
+      url = "http://downloads.openfabrics.org/verbs/${name}.tar.gz";
+      sha256 = "13w2j5lrrqxxxvhpxbqb70x7wy0h8g329inzgfrvqv8ykrknwxkw";
+  };
 
-  src = fetchurl {
-    url = "https://www.openfabrics.org/downloads/verbs/${name}.tar.gz";
-    sha256 = "13w2j5lrrqxxxvhpxbqb70x7wy0h8g329inzgfrvqv8ykrknwxkw";
+  drivers = {
+     libmlx4 = rec { 
+                     version="1.0.6";
+                     name="libmlx4-${version}"; 
+                     url="http://downloads.openfabrics.org/mlx4/${name}.tar.gz";
+                     sha256="f680ecbb60b01ad893490c158b4ce8028a3014bb8194c2754df508d53aa848a8";
+                   };
+     libmthca = rec { 
+                      version="1.0.6"; 
+                      name="libmthca-${version}"; 
+                      url="http://downloads.openfabrics.org/mthca/${name}.tar.gz";
+                      sha256="cc8ea3091135d68233d53004e82b5b510009c821820494a3624e89e0bdfc855c";
+                    };
   };
 
+in stdenv.mkDerivation rec {
+
+  version = verbs.version;
+  name = verbs.name;
+
+  srcs = [
+    (fetchurl {
+      url = verbs.url;
+      sha256 = verbs.sha256;
+    })
+
+    (fetchurl {
+      url = drivers.libmlx4.url;
+      sha256 = drivers.libmlx4.sha256;
+    })
+
+    (fetchurl {
+      url = drivers.libmthca.url;
+      sha256 = drivers.libmthca.sha256;
+    })
+ 
+  ];
+
+  sourceRoot=name;
+
+  # Install userspace drivers
+  postInstall=''
```

Comment:
I wonder it also. But as the plugins install configuration files into $libibverbs/etc/libibverbs.d and that libibverbs searchs into this path, there's a kind of chicken and egg problem: if we put plugins into a specific package, let's say libibverbs-plugins, the configuration files could be located into $libibverbs-plugins/etc/libibverbs.d. But libibverbs is a build dependency for the plugins and $libibverbs-plugins should be a build dependency of libibverbs to setup the path of the plugins configuration... If you know a solution for that problem, I would be glad to apply it :-)

Thank you very much for the other remarks, I'll check all this, and amend my commit as soon as possible.